src/parser/meta_parser.rs

```diff
@@ -90,6 +101,24 @@ pub fn parse_meta_delete_response(buf: &[u8]) -> Result<Option<(usize, MetaRespo
     }
 }
 
+pub fn parse_meta_arithmetic_response(
+    buf: &[u8],
+) -> Result<Option<(usize, MetaResponse)>, ErrorKind> {
+    let total_bytes = buf.len();
+    let result = parse_meta_arithmetic_data_value(buf);
+
+    match result {
+        Ok((remaining_bytes, response)) => {
+            let read_bytes = total_bytes - remaining_bytes.len();
+            Ok(Some((read_bytes, response)))
+        }
+        Err(nom::Err::Incomplete(_)) => Ok(None),
+        Err(nom::Err::Error(e)) | Err(nom::Err::Failure(e)) => {
+            Err(ErrorKind::Protocol(Some(e.code.description().to_string())))
+        }
+    }
+}
```

Comment:
These `parse_meta_*_response` methods can probably be refactored into a generic `parse_meta_response` closure that takes the appropriate value parser for each operation.